libs/shared/top-right-bar/src/lib/shared-top-right-bar.tsx

```diff
@@ -1,37 +1,117 @@
-import React, { ReactNode } from 'react';
-import Toolbar from '@material-ui/core/Toolbar';
-import IconButton from '@material-ui/core/IconButton';
-import AccountCircle from '@material-ui/icons/AccountCircle';
-import SettingsIcon from '@material-ui/icons/Settings';
-import { Link } from '@material-ui/core';
+import React, { useState, MouseEvent, ReactNode } from 'react';
 import './shared-top-right-bar.scss';
 
+import {
+  Toolbar,
+  IconButton,
+  MenuItem,
+  Menu,
+  Badge,
+  PopoverOrigin,
+  Link,
+} from '@material-ui/core';
+
+import {
+  AccountCircle,
+  Settings as SettingsIcon,
+  Notifications as NotificationsIcon,
+} from '@material-ui/icons';
+
+export interface Notifications {
+  _id: string;
+  title: string;
+  path: string;
+  timeStamp: string;
+  isRead: boolean;
+}
+
+export interface SharedTopMenuButtonProps {
+  children: ReactNode;
+  items: Notifications[];
+}
+
 export interface MenuIconLink {
   _id: string;
   label: string;
   path: string;
 }
 
 export interface SharedTopRightBarProps {
-  dataFromDb: {
-    settings: MenuIconLink;
-    profile: MenuIconLink;
-  };
-  children: ReactNode;
+  settings: MenuIconLink;
+  profile: MenuIconLink;
+  notifications: Notifications[];
+}
+
+export interface MenuProps {
+  vertical: string;
+  horizontal: string;
 }
 
+const menuPosition: PopoverOrigin = { vertical: 'top', horizontal: 'right' };
+
+const SharedTopMenuButton = (props: SharedTopMenuButtonProps) => {
+  const [menuOpen, setMenuOpen] = useState<boolean>(false);
+  const [anchorEl, setAnchorEl] = useState<HTMLElement>();
+
+  const handleMenu = (event: MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+    setMenuOpen(true);
+  };
+
+  const handleClose = () => {
+    setMenuOpen(false);
+  };
+
+  return (
+    <div>
+      <IconButton color="primary" onClick={handleMenu}>
+        {props.children}
+      </IconButton>
+
+      {props.items.length === 0 ? null : (
```

Comment:
" props.items.length === 0 ? null : (  " change to : " !!props.items.length && ( "